src/Domain/FunctionsOfTime/FunctionOfTimeHelpers.hpp

```diff
@@ -0,0 +1,72 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// \brief Helper struct and functions for FunctionsOfTime
+
+#pragma once
+
+#include <array>
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace FunctionOfTimeHelpers {
+/// \brief Stores info at a particlar time about a function and its derivatives
+///
+/// \details `MaxDerivPlusOne` template parameter is intended to be related to
+/// the maximum derivative that a FunctionOfTime is supposed to store. For a
+/// generic PiecewisePolynomial this will be the MaxDeriv + 1 (because we store
+/// the function as well as its derivatives).
+/// `StoreCoefs` template parameter indicates that the coefficients of the
+/// polynomial used for evaluation are stored instead of the polynomial itself.
+/// The coefficient of x^N is the Nth deriv rescaled by 1/factorial(N)
+template <size_t MaxDerivPlusOne, bool StoreCoefs = true>
```

Comment:
Would it be more convenient to take `MaxDeriv` as a parameter and add one when it is used internally?  I don't know what makes more sense.

Reply:
`MaxDeriv` usually comes from the templated parameter of a PiecewisePolynomial (also named `MaxDeriv`). Is it common to take a template parameter and pass it in as an argument to a constructor/function?

Reply:
Sorry, I wasn't clear.  I was proposing `template <size_t MaxDerivPlusOne, ...` change to `template <size_t MaxDeriv, ...` and then move the `+ 1`s around to compensate.

You can change this or not, whichever seems clearer to you.  I just want to make sure both options have been considered.

Reply:
Ah ok I understand. I think I'll leave it as is. When I see a `PiecewisePolynomial<2>` I think it's a 2nd order polynomial (`MaxDeriv=2`). When I see a `StoredInfo<2>` I think it's storing 2 things internally (`MaxDerivPlusOne=2`).

Reply:
OK.